### PR TITLE
ref(demo): refactore demo sources

### PR DESCRIPTION
**Pull request by danielsogl**
# PR Checklist
 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

I refactored the demo to use the new Angular HttpClient and reformated the code.

If you have questions why I removed the _ from private variables  you can read it here:  https://angular.io/guide/styleguide#style-03-04

### CODE REVIEW

File: demo/src/assets/json/versions.json [hunk 1]
```diff
@@ -1 +1 @@
-[]
+[];
```
Comment (valorkin) [inline demo/src/assets/json/versions.json]: semicolon in json file?
I suppose it should break parsing `JSON.parse('[];')`

**Review by valorkin — CHANGES_REQUESTED** [review]
I have sentiments to variables with underscore in name
Styleguide is a set of must and nice to haves
underscores is an optional point

thanks for fixing `prettier` help in code :)

One small fixed is required to get it merged
`;` in json should kill parsing of it

File: demo/src/assets/json/versions.json (hunk 1 above)
Comment (danielsogl) [reply demo/src/assets/json/versions.json]: Sure! I fixed it.

**Comment by danielsogl** [comment]
@valorkin fixed

**Review by valorkin — APPROVED** [review]

**Review by valorkin — APPROVED** [review]

**Comment by valorkin** [comment]
once again thanks for PR! :)

and sorry for the delay, was busy organizing https://ngtalks.io/ >.<

**Comment by danielsogl** [comment]
Thanks 😀

**Pull request merged**